strawberryfields/devicespec.py

```diff
@@ -107,6 +109,9 @@ def validate_parameters(self, **parameters: complex) -> None:
         Raises:
             ValueError: if an invalid parameter is passed
         """
+        if not self.gate_parameters:
+            return
```

Comment:
Does it makes sense to emit a warning message in this case?

Reply:
Perhaps, I'm not sure it's necessary though. Do you think so?

Reply:
My intuition is that if a caller invokes `validate_parameters()`, they expect some sort of validation to take place; if there is nothing to validate, they shouldn't call the method in the first place. The goal is to avoid a situation where a user _thinks_ a set of parameters have been validated when, in fact, no validation has taken place.

Put another way, it comes down to whether we think calling `validate_parameters()` is a mistake when there are no parameters. If the answer is "yes", we should raise a warning. If the answer is "no", the current behaviour is correct.

Reply:
I don't see calling `validate_parameters()` as a mistake even when there are no parameter restrictions. It simply means that all parameters values are valid. For example, a user might not know if there are any restrictions, and simply calls this function to make sure that the used parameters are valid.

This should perhaps rather be update to 
```suggestion
        if self.gate_parameters is None:
            return
```
so that an empty `gate_parameters` dict would mean that no parameters are valid (i.e., raising the `Parameter {p} not a valid parameter for this device` error below), although I think I'd prefer not raising a warning.

Reply:
Okay, sounds good!